### ISSUE TITLE
Update to native token when network is changed on Send/Swap screen

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#22572]
## Description 
Update to native token when network is changed on Send/Swap screen

## Steps to reproduce 

   1. Install 1.38.107
   2. Select Send tokens on Ethereum Mainnet
   3. Select a preset value or manually enter
   4. Select a different network say Polygon from the network dropdown
   5. Because of [brave/brave-core#12487](https://github.com/brave/brave-core/pull/12487), only input value is reset but the token still remains `ETH` instead of changing over to `MATIC`

## Actual result 

https://user-images.githubusercontent.com/17010094/165436960-928e1d3f-6181-4713-bac6-812594f25655.mp4

## Expected result
Should switch to native token when network is changed and not just reset the input values

## Issue reproduces how often 
Easy

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? NA
- Can you reproduce this issue with the current Play Store Beta version? Yes
- Can you reproduce this issue with the current Play Store Nightly version? Yes

## Device details
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): All
   - Android version: All
   
## Brave version
1.38.107

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? NA
- Does the issue resolve itself when disabling Brave Rewards?NA
- Is the issue reproducible on the latest version of Chrome? NA

### Additional information

cc: @Pavneet-Sing @wchen342

**Comment by deeppandya** [comment]
@srirambv will verify and close if all okay.

**Comment by srirambv** [comment]
This is still an issue on current master (1.42.43)

**Comment by wchen342** [comment]
Haven't got a change to fix it yet, will get on it soon.

**Comment by Pavneet-Sing** [comment]
[Info]: This needs to be fixed on android Dapps panel.

https://community.brave.com/t/android-brave-crashes-when-adding-a-network-in-brave-wallet/408504

**Comment by Pavneet-Sing** [comment]
@aman-m1 Thanks for the suggestion. We hide the icon due to security concerns though will pass on your suggestion.

**Comment by jamesmudgett** [comment]
@aman-m1 yes we can add something

**Comment by Pavneet-Sing** [comment]
This implementation on Panel is still pending on panel https://github.com/brave/brave-browser/issues/22572#issuecomment-1171468980, since this has been closed so created a new issue to fix it on panel.
https://github.com/brave/brave-browser/issues/24078

**Comment by srirambv** [comment]
Verification passed on the following devices running 1.43.74 x64 Beta build
- Verified steps from issue description
- Verified changing network resets the input value and updates the token to network's native token

Oppo Reno 5(Android 12) | Samsung Tab A (Android 10)
-- | --
<video src=https://user-images.githubusercontent.com/17010094/185556719-e90318e0-4ffe-49f0-af36-345121e9d3e4.mp4> | <video src=https://user-images.githubusercontent.com/17010094/185556775-ebdd8dc8-020d-49a0-874f-3397b252f895.mp4>
